api/src/main/java/org/openmrs/obs/handler/AbstractHandler.java

```diff
@@ -59,46 +60,30 @@ public AbstractHandler() {
 	 * @return File that the complex data should be written to
 	 */
 	public File getOutputFileToWrite(Obs obs) throws IOException {
-		// Get the title and remove the extension.
+		// Get the title and the extension
 		String title = obs.getComplexData().getTitle();
-		String extension = "." + getExtension(title);
-		
-		// If getExtension returns the title, there was no extension
-		if (getExtension(title).equals(title)) {
-			extension = "";
-		}
+		String extension = FilenameUtils.getExtension(title);
```

Comment:
@mks-d I'm not sure where else it is used either and I didn't want to break anything that relies on it. This method's return behaviour is different, so if any other method relied on this one, it could lead to bigger problems. I'd rather adjust code that uses the old method than "emulate" its behaviour with FilenameUtils, though I don't know if that's feasible. Can modules access these methods, or would dependencies be isolated to openmrs-core and therefore easy to find?

Reply:
>"emulate" its behaviour with FilenameUtils

This sounds like a great idea ;-) You know by now that I love `FilenameUtils`!

I think it is unlikely albeit possible that modules rely on this specific method. So rather be safe than sorry indeed.

Reply:
Yeah ;D. Alright, I'll rewrite the old getExtension tomorrow to provide the same returns for the same inputs but internally use FilenameUtils.